go/consensus/cometbft/apps/staking/slashing.go

```diff
@@ -69,7 +69,7 @@ func onEvidenceByzantineConsensus(
 	// validator from being scheduled in the next epoch.
 	if penalty.FreezeInterval > 0 {
 		var epoch beacon.EpochTime
-		epoch, err = ctx.AppState().GetEpoch(context.Background(), ctx.BlockHeight()+1)
+		epoch, err = ctx.AppState().GetEpoch(context.Background(), ctx.BlockHeight())
```

Comment:
Why this change given that semantics of `GetEpoch` has not changed? It seems like this is consensus-breaking.

Reply:
I have also updated the `GetCurrentEpoch` code, so that it now similarly checks the provided context, and uses the appropriate height based on that, which should ensure consistent code with existing.

Reply:
Thanks for pointing this out, so I looked a bit more into it. I have added a comment for now. I also updated the `EpochChanged` fn in `applicationState` to similarly use the correct height, although that one is currently never used outside of abciCtx.